### PR TITLE
Missing class-specific features (class_features_data)

### LINKED ISSUE #19: Missing class-specific features
These are currently not stored in the Character object, although they exist in the class data.

```python
>>> from pprint import pprint
>>> from dnd_character.SRD import SRD_class_levels
>>> for classs in SRD_class_levels:
...     print(f"SRD_class_levels[{classs}][0]['class_specific'] = ")
...     pprint(SRD_class_levels[classs][0]['class_specific'])
...     print()
... 

SRD_class_levels[barbarian][0]['class_specific'] = 
{'brutal_critical_dice': 0, 'rage_count': 2, 'rage_damage_bonus': 2}

SRD_class_levels[bard][0]['class_specific'] = 
{'bardic_inspiration_die': 6,
 'magical_secrets_max_5': 0,
 'magical_secrets_max_7': 0,
 'magical_secrets_max_9': 0,
 'song_of_rest_die': 0}

SRD_class_levels[cleric][0]['class_specific'] = 
{'channel_divinity_charges': 0, 'destroy_undead_cr': 0}

SRD_class_levels[druid][0]['class_specific'] = 
{'wild_shape_fly': False, 'wild_shape_max_cr': 0, 'wild_shape_swim': False}

SRD_class_levels[fighter][0]['class_specific'] = 
{'action_surges': 0, 'extra_attacks': 0, 'indomitable_uses': 0}

SRD_class_levels[monk][0]['class_specific'] = 
{'ki_points': 0,
 'martial_arts': {'dice_count': 1, 'dice_value': 4},
 'unarmored_movement': 0}

SRD_class_levels[paladin][0]['class_specific'] = 
{'aura_range': 0}

SRD_class_levels[ranger][0]['class_specific'] = 
{'favored_enemies': 1, 'favored_terrain': 1}

SRD_class_levels[rogue][0]['class_specific'] = 
{'sneak_attack': {'dice_count': 1, 'dice_value': 6}}

SRD_class_levels[sorcerer][0]['class_specific'] = 
{'creating_spell_slots': [], 'metamagic_known': 0, 'sorcery_points': 0}

SRD_class_levels[warlock][0]['class_specific'] = 
{'invocations_known': 0,
 'mystic_arcanum_level_6': 0,
 'mystic_arcanum_level_7': 0,
 'mystic_arcanum_level_8': 0,
 'mystic_arcanum_level_9': 0}

SRD_class_levels[wizard][0]['class_specific'] = 
{'arcane_recovery_levels': 1}
```

### PR DESCRIPTION
**Pull request by gergo-szabo**
Discussed in #19 

Not discussed in #19:
- `get_class_features_data` function has two parameter `class_name` and `level`. If multi class characters will be possible then this function will only need a wrapper which gets class levels and concats the results.
- During leveling a simple call for `get_class_features_data` is not enough. Counter types shouldn't be reseted!

There is a huge amount of counter type mechanics. I double checked everything but it is possible I have missed something. 
I created a simple test but it doesn't cover every edge case.

Closes #19

### CODE REVIEW

**Comment by gergo-szabo** [comment]
Sorry about the spam, long day... I will finish it on Friday

File: dnd_character/character.py [hunk 1]
```diff
@@ -842,3 +863,171 @@ def get_maximum_hp(hd: int, level: int, constitution: int) -> int:
...
+
+    def get_class_features_data(self, class_name, class_level):
+        """
+        Creates a dict for class features. Assumes a fully rested character!
+        """
+        if class_level is None or class_name is None:
+            return None
+
+        # Load level appropriate class features
+        data = SRD_class_levels[class_name.lower()][class_level - 1]["class_specific"]
```
Comment (tassaron) [inline dnd_character/character.py:875]:
```suggestion
        data = SRD_class_levels[class_name.lower()][class_level - 1]["class_specific"].copy()
```

**Comment by tassaron** [comment]
Thanks so much. No worries about the 'spam', we can squash some commits down later. I appreciate your work

**Comment by gergo-szabo** [comment]
Character init has a `class_features_data` parameter. It is needed for serialization but the init doesn't use it (`class_features_data` variable has a fix value until level or class changes).

`test_class_features_data.py` tests for:
- no class -> None
- no level -> default level is 1, get the correct data
- level correct data for 9th level barbarian
- ability score dependent class feature in case of a bard (inspiration count)
- "days_since" and "avalable" type of features in case of a wizard
There are other mechanisms but those are fairly simple.

**Comment by tassaron** [comment]
Since `class_features_data` is something internally managed by the Character class, I put the underscore signifying that it is "private". This prevents it from being serialized so we don't need it for `init`

I didn't put a setter method so that might be needed if the user is expected to manually modify the `available` values... maybe we could do some validation. But this is fine for now, I think.

I am curious why "[7 day cooldown](https://github.com/gergo-szabo/dnd-character/blob/class_features/dnd_character/character.py#L987)" = 999 though? What is the 999?

**Comment by gergo-szabo** [comment]
I am not sure if I found every ability with cooldown. The largest cooldown I would expect is 1 year. So I choose 999 day to signal all abilities are available.
We could count the days until ability available if you want to. But then we need some logic for keeping it >= 0.

**Comment by tassaron** [comment]
Okay. Your comments of '1 day' and '7 days' are referring to the cooldown that could be implemented. The 999 is just a suitably large number that the ability wouldn't be cooled down after reset. I think I understand.

I will change the comments a bit and merge shortly. Thank you :)
